transform/instance.go

```diff
@@ -32,18 +33,19 @@ type instanceTransformer interface {
 	child() instanceTransformer // Arrays return a child object all others nil
 	path() string
 	selectChild(string) instanceTransformer // This returns nil for everything except objects
-	transform(interface{}, pathModifier) (interface{}, error)
+	transform(interface{}, pathModifier, string) (interface{}, error)
 }
 
-// arrayTransformer represents a JSON instance of type array.
+// arrayTransformer represents a JSON instance type array in the case of a JSON transform or an array of xmlquery.Node in the case of an XML transform.
+// in both cases the output will be JSON
 type arrayTransformer struct {
 	childTransformer instanceTransformer
 	defaultValue     []interface{}
 	jsonPath         string
 	transforms       *transformInstructions
 }
 
-func newArrayTransformer(path, transformIdentifier string, raw json.RawMessage) (*arrayTransformer, error) {
+func newArrayTransformer(path, transformIdentifier string, raw json.RawMessage, _ string) (*arrayTransformer, error) {
```

Comment:
Hmm, I was thinking if you set the type into the arrayTransformer struct here you don't have to pass it in for other functions just in the new part.